src/sentry/api/endpoints/organization_spans_aggregation.py

```diff
@@ -0,0 +1,230 @@
+import hashlib
+from collections import defaultdict, namedtuple
+from typing import Dict, List, TypedDict, Union
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+from snuba_sdk import Column, Function
+
+from sentry import features
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import NoProjects, OrganizationEventsEndpointBase
+from sentry.models import Organization
+from sentry.search.events.builder import QueryBuilder
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+EventSpan = namedtuple(
+    "EventSpan",
+    [
+        "span_id",
+        "is_segment",
+        "parent_span_id",
+        "group",
+        "group_raw",
+        "description",
+        "start_ms",
+        "duration",
+        "exclusive_time",
+    ],
+)
+
+
+class EventSpans(TypedDict):
+    transaction_id: str
+    spans: List[EventSpan]
+
+
+NULL_GROUP = "00"
+
+
+@region_silo_endpoint
+class OrganizationSpansAggregationEndpoint(OrganizationEventsEndpointBase):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    aggregated_tree: Dict[str, Dict[str, Union[str, int, float]]] = {}
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not features.has("organizations:starfish-view", organization, actor=request.user):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        transaction = request.query_params.get("transaction", None)
+        if transaction is None:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST, data={"details": "Transaction not provided"}
+            )
+
+        builder = QueryBuilder(
```

Comment:
Could we use the `SpansIndexedQueryBuilder` here

Reply:
addressed here! https://github.com/getsentry/sentry/pull/56291/commits/1fbbfe05516aba2dd890cb846b44a7af04a3de77#diff-1df1994c8fddccf95629e8b491f1ff17c43534b3634dd396256fc448124d7cddR67